content/en/docs/collector/building/_index.md

```diff
@@ -9,3 +9,22 @@ also by custom components, that you develop and build on your own. Here you will
 find instructions on how to build some of those components. For additional
 details take a look into the documents contained within the
 [opentelemetry-collector-contrib repository](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/README.md).
+
+
+There are several [classes](https://github.com/open-telemetry/opentelemetry-collector/blob/main/cmd/mdatagen/main.go#L184) of opentelemetry components, such as
+- `receivers` (scrapers and listeners to ingest data)
+- `exporters` (ways to export data to vendors or other tooling)
+- `processors` (ways to process data in a pipeline)
+- `connectors` (ways to [connect pipelines](./connector/) and form a [DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph) for your data pipeline)
+- `extensions` (ways to [augment the collector](https://github.com/open-telemetry/opentelemetry-collector/blob/main/extension/README.md), such as [health checks](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/healthcheckextension/README.md))
+- `cmd` (commands)
+- `pkg` (packages)
+
+Most components are registered via [`mdatagen`](https://github.com/open-telemetry/opentelemetry-collector/tree/main/cmd/mdatagen), a tool which will auto generate some code specifying the adoption status (stability) of your component along with documentation and component specific information. Most significantly, `mdatagen` configuration is used to configure the out-of-the-box metrics for scraping receivers, and which are enabled by default for collection.
```

Comment:
add a cspell ignore for `mdatagen` to the front matter of that page (look into other markdown files how this can be accomplished)